core/Makefile

```diff
@@ -18,10 +18,9 @@ $(shell mkdir -p $(DEPDIR)/hooks >/dev/null)
 # 'clang' or 'g++'
 CXXCOMPILER := $(shell expr $(word 1, $(shell $(CXX) --version)) : '\(clang\|g++\)')
 
-# e.g., 4.9.3 -> 40903. For clang it is always 40201
-CXXVERSION := $(shell $(CXX) -dumpversion | sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g' -e 's/^[0-9]\{3,4\}$$/&00/')
+CXXVERSION := $(shell $(CXX) -dumpversion)
 
-ifeq "$(shell expr $(CXXCOMPILER) = g++ \& $(CXXVERSION) \< 40800)" "1"
+ifneq "$(shell echo -e '$(CXXVERSION)\n4.7' | sort -V | head -n1)" "4.7"
```

Comment:
I don't understand how it works. `clang -dumpversion` is supposed to report `4.2.1` no matter what its real version is (this is why the previous code checked whether it is `g++`) first. So in theory build with clang should fail, but somehow it passed CI?